qatzip/decompress.go

```diff
@@ -195,19 +210,24 @@ func (z *Reader) Read(p []byte) (n int, err error) {
 			in, out, len(z.inputBuf), z.inputBufOffset, z.inputBufRead, len(z.outputBuf), err)
 
 		if err != nil {
-			if err == ErrBuffer {
-				// expand output buffer
-				// TODO grow to a maximum size
+			if err == ErrBuffer && !isPartialJob {
+				// Grow output buffer, only in the case that it was not large enough for QAT to output any data.
+				// If QAT does output data, we save a copy by doing it after the output buffer draining.
+				
 				t1 = time.Now().UnixNano()
-				z.bufferGrowth *= 2
-				z.traceLogf(Med, "[expand output buffer] obl:%v -> %v", len(z.outputBuf), remainder+z.bufferGrowth)
-				z.outputBuf = make([]byte, remainder+z.bufferGrowth)
+				newLength := len(z.outputBuf)*z.bufferGrowth
+				if newLength > MaxBufferLength {
+					newLength = MaxBufferLength
+				}
```

Comment:
There is an edge case we need to test for when the input size is 2GB and it has no way to expand it could cause a infinite loop. You need to just return an error in that case.